sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -53,7 +53,7 @@ object RDDConversions {
     data.mapPartitions { iterator =>
       val numColumns = outputTypes.length
       val mutableRow = new GenericInternalRow(numColumns)
-      val converters = outputTypes.map(CatalystTypeConverters.createToCatalystConverter)
+      val converters = outputTypes.map(CatalystTypeConverters.createToCatalystConverter).toArray
```

Comment:
shall we use `RowEncoder` here?

Reply:
It has been modified, and the performance is the same as converting to arrays.